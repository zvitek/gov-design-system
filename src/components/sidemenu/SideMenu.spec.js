import { shallowMount } from '@vue/test-utils'
import GovSideMenu from '@components/sidemenu/SideMenu'

let wrapper

describe('GovSideMenu', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovSideMenu)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovSideMenu')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should be complex', () => {
        wrapper.setProps({
            complex: true
        })
        expect(wrapper.classes()).toContain('gov-sidenav--complex')
    })

    it('should have default slot', () => {
        const slotDefault = '<div> Content </div>'
        wrapper = shallowMount(GovSideMenu, {
            slots: {
                default: slotDefault
            }
        })
        expect(wrapper.find('.gov-sidenav :first-child').html()).toBe(slotDefault)
    })
})
