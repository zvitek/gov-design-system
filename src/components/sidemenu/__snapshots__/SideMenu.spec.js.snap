// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovSideMenu render correctly 1`] = `<nav class="gov-sidenav"></nav>`;
